jira/resources.py

```diff
@@ -763,14 +764,46 @@ def __init__(
             self._parse_raw(raw)
         self.raw: Dict[str, Any] = cast(Dict[str, Any], self.raw)
 
-    def update(self, fields=None, async_=None, jira=None, body="", visibility=None):
-        """Update a comment"""
-        data = {}
+    # mypy complains about the signature as the order of the parameters is different.
+    # If we follow mypy here we will break backwards compatibility with version <= 3.2 which is probably not advised
+    @typing.no_type_check
+    def update(
+        self,
+        fields: Optional[Dict[str, Any]] = None,
+        async_: Optional[bool] = None,
+        jira: "JIRA" = None,
```

Comment:
more for myself, but why the quotes around JIRA?
I didn't see such a variant on https://docs.python.org/3/library/typing.html

Reply:
This is because we only import the `JIRA` class when type checking, and current Python versions actually try to evaluate the type hint.

Some more reading here: https://mypy.readthedocs.io/en/stable/runtime_troubles.html#annotation-issues-at-runtime